controllers/etcd_controller.go

```diff
@@ -978,6 +994,18 @@ func (r *EtcdReconciler) getMapFromEtcd(etcd *druidv1alpha1.Etcd) (map[string]in
 		sharedConfigValues["autoCompactionRetention"] = etcd.Spec.Common.AutoCompactionRetention
 	}
 
+	etcdConfigMountPath := "/var/etcd/config/"
+	initialCluster := fmt.Sprintf("%s=http://0.0.0.0:2380", etcd.Name)
+	if statefulsetReplicas > 1 {
+		etcdConfigMountPath = "/var/etcd/template/config/"
+		// form initial cluster
+		initialCluster = ""
+		for i := 0; i < statefulsetReplicas; i++ {
+			initialCluster = initialCluster + fmt.Sprintf("%s=http://%s-%d.%s:2380,", etcd.Name, etcd.Name, i, etcd.Name)
+		}
+	}
+
+	initialCluster = strings.Trim(initialCluster, ",")
```

Comment:
Can you please make this into a helper function which then can be unit tested to see if the initial cluster string is being generated properly in all the cases?